lib/segment/src/index/field_index/numeric_index/mod.rs

```diff
@@ -325,6 +340,46 @@ impl<T: Encodable + Numericable + Default> NumericIndexInner<T> {
     pub fn values_is_empty(&self, idx: PointOffsetType) -> bool {
         self.values_count(idx) == 0
     }
+
+    pub fn values_by_key(&self, key: &T) -> Box<dyn Iterator<Item = PointOffsetType> + '_> {
+        let start = Bound::Included(Point::new(*key, PointOffsetType::MIN));
+        let end = Bound::Included(Point::new(*key, PointOffsetType::MAX));
+        match &self {
+            NumericIndexInner::Mutable(mutable) => Box::new(mutable.values_range(start, end)),
+            NumericIndexInner::Immutable(immutable) => Box::new(immutable.values_range(start, end)),
+        }
+    }
+
+    /// Tries to estimate the amount of points for a given key.
+    pub fn estimate_points(&self, key: &T) -> usize {
```

Comment:
I guess some unit tests for numeric index index changes are suitable. Not required for this PR IMHO but we need python integration test at least